BioCro/DARPA/upload_params_data/BETY/Compare_BETY_to_upload.Rmd

```diff
@@ -0,0 +1,61 @@
+---
+title: "Compare BETY and upload_params_data"
+---
+  
+  This script compares the results of what is in BETY to what has been uploaded to it, in an effort to identify mistakes or duplicates. 
+
+Data from BETY is selecte with the following SQL query: select * from traits where treatment_id in (9000000015, 9000000016, 9000000017, 9000000018, 9000000019, 9000000020)
+The results are saved as csv in the upload_params_data/BETY folder as the most recent 'update_at' date. 
+
+Load libraries and both sets of data
+```{r}
+options(scipen = 999)
+library(dplyr)
+library(arsenal)
+#import newest BETY file
+filename <- max(as.numeric(substr(list.files(pattern = "csv"), 1,8)))
+in_bety <- read.csv(paste0(filename, ".csv"), header = T)
+#import full upload_params_data data file
+in_upload <- read.csv("../phys_parameters.csv",  header = T)
+```
+
+Convert the treatment and variable ID columns to match
+```{r}
+bety <- in_bety %>% 
+  mutate(date = as.POSIXct(as.character(date), format = "%d/%m/%Y %H:%M:%S"),
+         treatment = case_when(treatment_id == "9000000015" ~ "regular night temperature",
+                               treatment_id == "9000000016" ~ "high night temperature",
+                               treatment_id == "9000000017" ~ "high night",
+                               treatment_id == "9000000018" ~ "greenhouse",
+                               treatment_id == "9000000019" ~ "outdoor 5 cm density",
+                               treatment_id == "9000000020" ~ "outdoor JollyG soil"),
+         variable = case_when(variable_id == "7" ~ "leaf_respiration_rate",
+                              variable_id == "39" ~ "quantum_efficiency",
+                              variable_id == "404" ~ "stomatal_slope.BB",
+                              variable_id == "4" ~ "Vcmax",
+                              variable_id == "2000000027" ~ "theta",
+                              variable_id == "38" ~ "cuticular_cond"),
+         site = case_when(site_id == "9000000005" ~ "Donald Danforth Plant Science Center Greenhouse",
+                          site_id == "9000000004" ~ "Donald Danforth Plant Science Center Growth Chamber",
+                          site_id == "9000000006" ~ "Donald Danforth Plant Science Center Outdoor"),
+         SE = stat,
+         entity = entity_id) %>%
+  select(date, site, treatment, variable, mean, n, SE, entity) %>%
+  arrange(site, treatment, variable, entity) %>%
+  select(-date, -entity)
+```
+
+Create matching version of upload data
+```{r}
+upload <- in_upload %>%
+  select(site, treatment, variable, mean, n, SE, entity )%>%
+  arrange(site, treatment, variable, entity) %>%
+  select(-entity)
+```
+
+Compare both data versions
+```{r}
+compare(bety, upload)
```

Comment:
When I ran this, this is what is returned: 
```
Compare Object

Function Call: 
compare.data.frame(x = bety, y = upload)

Shared: 7 variables and 182 observations.
Not shared: 0 variables and 0 observations.

Differences found in 1/3 variables compared.
3 variables compared have non-identical attributes.
```
The two do look identical from some spot checking, what does the last bit mean about "non-identical attributes"? If that message is of no concern, it would be worthwhile to expand on that in the text below a bit.

Reply:
Thanks for pointing this out. There were some differences in character vs. factor, and a couple other things. Now all the correct lines read 0, but of course I'll have to update this after the add_SLA PR gets merged, haha!

Reply:
This will probably have to continually be updated as data is added, including updating the Bety dump. It's neverending!